### PR TITLE
Fix running AD through `jaxsim.math.Rotation.from_axis_angle`

### PR DESCRIPTION
**Pull request by diegoferigo**
In these days, I tried to solve an unconstrained non-linear optimization problem using the JaxSim functions to construct the cost function. I used the [`L-BFGS`](https://optax.readthedocs.io/en/latest/_collections/examples/lbfgs.html#l-bfgs-solver) solver from optax, and it works flawlessly :tada:

What I'm trying to solve is quite usual in AMI. I want to compute the initial configuration of a humanoid robot such that the knees and the elbows are slightly bent, the soles of the feet are in contact, and the feet are slightly more distant w.r.t. the configuration with zero joint positions. The optimized variables are the joint positions and the base height. The corresponding optimization problem computes FK in the cost function (exploiting AD support through the JaxSim function).

I had to fix the gradient propagation as in this PR to prevent having problems either with zero division in the jitted function  or the same problem in the gradient computed with AD. The trick I used it the _double-where_ workaround, as described in:

- https://jax.readthedocs.io/en/latest/faq.html#gradients-contain-nan-where-using-where
- https://github.com/jax-ml/jax/issues/5039#issuecomment-735430180
- https://github.com/jax-ml/jax/issues/1052#issuecomment-514083352

----
📚 Documentation preview 📚: https://jaxsim--277.org.readthedocs.build//277/

### CODE REVIEW

**Comment by diegoferigo** [comment]
Note: probably a similar trick can be used to remove the small tolerances that we always had in `SoftContacts`:

https://github.com/ami-iit/jaxsim/blob/35cf33692164fe86872a2f3c9e557124353c5013/src/jaxsim/rbda/contacts/soft.py#L288-L292

**Review by flferretti — APPROVED** [review]
Thanks Diego, LGTM!
